x-pack/plugins/profiling/public/embeddables/stack_traces/stack_traces.tsx

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { StackTracesDisplayOption, TopNType } from '@kbn/profiling-utils';
+import React, { useState } from 'react';
+import { groupSamplesByCategory } from '../../../common/topn';
+import { useProfilingDependencies } from '../../components/contexts/profiling_dependencies/use_profiling_dependencies';
+import { StackTraces as StatelessStackTraces } from '../../components/stack_traces';
+import { useAsync } from '../../hooks/use_async';
+
+interface Props {
+  type: TopNType;
+  kuery: string;
+  rangeFrom: number;
+  rangeTo: number;
+  onClick: (category: string) => void;
+  onChartBrushEnd: (range: { rangeFrom: string; rangeTo: string }) => void;
+}
+
+export function StackTraces({ type, kuery, rangeFrom, rangeTo, onClick, onChartBrushEnd }: Props) {
+  const {
+    services: { fetchTopN },
+  } = useProfilingDependencies();
+  const [displayOption, setDisplayOption] = useState(StackTracesDisplayOption.StackTraces);
+
+  const rangeFromSec = rangeFrom / 1000;
+  const rangeToSec = rangeTo / 1000;
+
+  const state = useAsync(
```

Comment:
As this component fetches the data itself instead of taking it from the props, does it also handle error and empty states?

Reply:
I added them here: https://github.com/elastic/kibana/pull/173905/commits/155b9b0de1aa3a2d5fec614bb413ca1fac1f5cac

For now I copied the same components you created on the Infra side. We can improve that later. Maybe it would be a good idea to encapsulate this logic on each Embeddable, so they know how to display data/error/empty states.

Reply:
Nice, thank you for addressing this!